pylint/checkers/classes/class_checker.py

```diff
@@ -1775,15 +1796,15 @@ def _is_type_self_call(self, expr: nodes.NodeNG) -> bool:
         )
 
     @staticmethod
-    def _is_classmethod(func):
+    def _is_classmethod(func: nodes.FunctionDef) -> bool:
```

Comment:
Shouldn't this be `LocalsDictNodeNG`? Otherwise the check on 1802 makes no sense.